bin/postgresql-upgrade.in

```diff
@@ -163,6 +163,27 @@ if [ ! -x "$upgradefrom_engine/postgres" ]; then
     exit 1
 fi
 
+# Check for data_directory entry in config file
+# valid entry means that custom PGDATA path is present which is not supported
+# BZ (#1935301)
+conffile="$option_datadir/postgresql.conf"
+
+test -r "$conffile" || {
+    error "config file $conffile is not readable or does not exist"
+    die "Old cluster in '$data' does not seem to be initialized"
+}
+
+sp='[[:space:]]'
+sed_expr_pgdata="s/^$sp*data_directory$sp*=$sp[\'|\"]\(.*\+\)[\'|\"].*/\1/p"
```

Comment:
Wouldn't `s/^$sp*data_directory$sp*=` be enough here? I'm thinking whether the rest is not just a potential risk for not matching the line.. Or maybe I miss the point of getting the value itself.

Reply:
Yes, it will extract data_directory value. Previous version had minor issue. New regex can deal with all/most of the cases:

```
data_directory = '/example/here_it_is.conf'
data_directory='/example/here_it_is.conf'
data_directory= '/example/here_it_is.conf'
data_directory= '/example/here_it_is.conf'
 data_directory = '/example/here_it_is.conf'
```

```
data_directory = /example/here_it_is.conf
data_directory=/example/here_it_is.conf
data_directory= /example/here_it_is.conf
data_directory= /example/here_it_is.conf
 data_directory = /example/here_it_is.conf
```

```
data_directory = "/example/here_it_is.conf"
data_directory="/example/here_it_is.conf"
data_directory= "/example/here_it_is.con"'
data_directory= "/example/here_it_is.con"'
 data_directory = "/example/here_it_is.conf"
```